rmldatatfbbox/rmldatatfbbox/core.py

```diff
@@ -5,77 +5,158 @@
 Core command group and commands for TF Bounding Box dataset plugin.
 """
 import click 
-
-### OPTIONS ###
-
+import sys
+import io
+import importlib
+import shutil
+from pathlib import Path
+import tensorflow as tf
+from rmldatatfbbox.utils.helpers import (construct_all, write_label_map, export_as_TFExample)
+from ravenml.utils.question import user_input, cli_spinner
+from ravenml.data.helpers import default_filter_and_load
+from ravenml.data.write_dataset import write_dataset, write_metadata
+from ravenml.data.options import pass_create
+from ravenml.data.interfaces import CreateInput, CreateOutput
 
 ### COMMANDS ###
-@click.command(help='TensorFlow Object Detection with bounding boxes.')
+@click.group(help='Dataset creation with bounding box info')
 @click.pass_context
 def tf_bbox(ctx):
     pass
+
+@tf_bbox.command(help='Create a dataset.')
+@pass_create
+@click.pass_context
+def create(ctx: click.Context, create: CreateInput):
+
+    cli_spinner("Importing TensorFlow...", _import_od)
+
+    config = create.config.get("plugin")
+
+    # Optional Information from interface (just contains plugin)
+    # metadata = create.plugin_metadata
```

Comment:
This metadata field only contains the plugin's name so not really useful right now

Reply:
See the `__init__` in `TrainInput` (https://github.com/autognc/ravenML/blob/master/ravenml/train/interfaces.py#L44) for things to add to metadata. The metadata should be created by ravenML core and passed to the plugin for the plugin to add anything else it wants. Things like author name, dataset type, etc should be set by the core